doc/dev/transforms.rst

```diff
@@ -100,6 +100,117 @@ classical instructions as well as side-effecting operations (e.g., print) can al
 same representation.
 
 
+Writing and running your first Catalyst pass
+============================================
+
+If this is your first time writing MLIR or LLVM passes, the boilerplate can be quite overwhelming. 
+Let's first set up the various boilerplate items required to register and run a new pass. 
+
+We'll create an empty pass in the ``Catalyst`` dialect that just prints out hello world to stdout. The full codelisting of the pass already exists in Catalyst as ``MyHelloWorldPass`` in the directories ``mlir/include/Catalyst/Transforms`` and ``mlir/lib/Catalyst/Transforms``, and we will go over it here. Note that the ``mlir/include`` (and ``mlir/lib``) directories consists of all the available dialects, so if you want to write a new pass in another dialect, it should be added to the subdirectory of that dialect. 
+
+The first thing to do is to create the pass object in the tablegen ``mlir/include/Catalyst/Transforms/Passes.td``:
+
+.. code-block::
+
+    def MyHelloWorldPass : Pass<"my-hello-world"> {
+        let summary = "An empty pass boilerplate that prints out hello world.";
+
+        let constructor = "catalyst::createMyHelloWorldPass()";
+    }
+
+When the dialect is built, this tablegen def will be built to a C++ file ``Catalyst/Transforms/Passes.h.inc``, containing the newly defined object called ``MyHelloWorldPass``, alongside the various necessary boilerplate methods in the MLIR infrastructure. Tablegen is designed such that we don't have to write all that boilerplate ourselves. 
```

Comment:
```suggestion
When the dialect is built, this tablegen def will be built to a C++ file ``mlir/build/include/Catalyst/Transforms/Passes.h.inc``, containing the newly defined object called ``MyHelloWorldPassBase``, alongside the various necessary boilerplate methods in the MLIR infrastructure. Tablegen is designed such that we don't have to write all that boilerplate ourselves. 
```

perhaps clear to most but it took me a while to find them as there is also just `mlir/include/Catalyst/Transforms/`

Reply:
This is valuable to add!